### PR TITLE
Plots selflearn

### PR DESCRIPTION
**Pull request by pp-mo**
Suggested general improvements to Iris course section 4 (plotting).

Unfortunately I got rid of all the outputs, which again creates a lot of noise.
Can we discuss @lbdreyer ?

### CODE REVIEW

File: course_content/iris_course/4.Cube_Plotting.ipynb [hunk 1]
```diff
@@ -148,8 +151,103 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "----\n",
+    "\n",
+    "Almost all the Iris plot methods have both `iplt` and `qplt` versions.  \n",
+    "Also, most of these have the same names as similar methods in `matpltloib.pyplot`."
```
Comment (lbdreyer) [inline course_content/iris_course/4.Cube_Plotting.ipynb]: Could have links to the documentation for [iris.plot](https://scitools.org.uk/iris/docs/latest/iris/iris/plot.html) and iris.quickplot to encourage them to have a look at what is available.

File: course_content/iris_course/solutions/iris_exercise_4.1c [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+# Iris course sample answer 4.1c
+# blockplot a 2d portion of "airtemps"
```
Comment (lbdreyer) [inline course_content/iris_course/solutions/iris_exercise_4.1c]: This should be a `filled contour plot ...`

File: course_content/iris_course/solutions/iris_exercise_4.1b [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# Iris course sample answer 4.1b
+# make a colour blockplot of "airtemps"
+
+try:
+    qplt.pcolormesh(airtemps)
+    plt.show()
+except Exception as err:
+    print('Error occurred:\n  "{}"'.format(err))
+
+print("NOTE: the 'qplt.pcolormesh' method requires a 2d cube.")
```
Comment (lbdreyer) [inline course_content/iris_course/solutions/iris_exercise_4.1b]: The suggested activity tells the user to do a `qplt.contourf` but this answer uses pcolormesh.
As pcolormesh gets introduced later, I think it makes sense to use contourf.

File: course_content/iris_course/solutions/iris_exercise_4.3b [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+# Iris course sample answer 4.3b
+# a block plot (pcolormesh) map in its native projection
+
+qplt.contourf(airtemps[0])
```
Comment (lbdreyer) [inline course_content/iris_course/solutions/iris_exercise_4.3b]: This should be pcolormesh

File: course_content/iris_course/4.Cube_Plotting.ipynb [hunk 5]
```diff
@@ -29,27 +29,11 @@
...
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "2.3.0dev0\n"
-     ]
-    }
-   ],
-   "source": [
+    "import iris\n",
     "print(iris.__version__)"
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply course_content/iris_course/4.Cube_Plotting.ipynb]: Ok, I agree it's mostly pointless.

File: course_content/iris_course/4.Cube_Plotting.ipynb (hunk 1 above)
Comment (pp-mo) [reply course_content/iris_course/4.Cube_Plotting.ipynb]: :+1: will do

**Comment by pp-mo** [comment]
Rebased. Should now be ok.

**Comment by pp-mo** [comment]
Think that should do it.
Please re-review @lbdreyer

File: course_content/iris_course/solutions/iris_exercise_4.1a [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+# Iris course sample answer 4.1a
```
Comment (lbdreyer) [inline course_content/iris_course/solutions/iris_exercise_4.1a]: Now that plotting in section 5 these should be updated (and the filenames)

**Comment by pp-mo** [comment]
OK, done that too.
How does it look now @lbdreyer ?

**Comment by lbdreyer** [comment]
Thanks @pp-mo

I really like this! It convinces me that the changes we have made to the course are a great improvement!

**Comment by lbdreyer** [comment]
Just spotted that the numbering of the headers are need to be updated. 
I'll push up a quick PR to fix this!

**Comment by lbdreyer** [comment]
> Just spotted that the numbering of the headers are need to be updated.
> I'll push up a quick PR to fix this!

See #161

**Pull request merged**